examples-gallery/plot_crane_moving_a_load.py

```diff
@@ -0,0 +1,337 @@
+
+t0, tf = 0.0, duration
+
+
+# %%
+# Set up the objective function and its gradient.
+def obj(free):
+    """Minimize h, the time interval between nodes."""
+    return free[-1]
+
+
+def obj_grad(free):
+    grad = np.zeros_like(free)
+    grad[-1] = 1.0
+    return grad
+
+
+# %%
+# Starting and final location of the load.
+starting_location = 0.0
+ending_location = 15.0
+
+# %%
+# Set the initial and final states to form the instance constraints.
+initial_state_constraints = {
+    xc: starting_location,
+    xl: starting_location,
+    yl: -par_map[l],
+    q: 0.0,
+    uxc: 0.0,
+    uxl: 0.0,
+    uyl: 0.0,
+    u: 0.0,
+}
+
+final_state_constraints = {
+    xc: ending_location,
+    xl: ending_location,
+    yl: -par_map[l],
+    q: 0.0,
+    uxc: 0.0,
+    uxl: 0.0,
+    uyl: 0.0,
+    u: 0.0,
+}
+
+instance_constraints = (
+    xc.subs({t: t0}) - initial_state_constraints[xc],
+    xl.subs({t: t0}) - initial_state_constraints[xl],
+    yl.subs({t: t0}) - initial_state_constraints[yl],
+    q.subs({t: t0}) - initial_state_constraints[q],
+    uxc.subs({t: t0}) - initial_state_constraints[uxc],
+    uxl.subs({t: t0}) - initial_state_constraints[uxl],
+    uyl.subs({t: t0}) - initial_state_constraints[uyl],
+    u.subs({t: t0}) - initial_state_constraints[u],
+    xc.subs({t: tf}) - final_state_constraints[xc],
+    xl.subs({t: tf}) - final_state_constraints[xl],
+    yl.subs({t: tf}) - final_state_constraints[yl],
+    q.subs({t: tf}) - final_state_constraints[q],
+    uxc.subs({t: tf}) - final_state_constraints[uxc],
+    uxl.subs({t: tf}) - final_state_constraints[uxl],
+    uyl.subs({t: tf}) - final_state_constraints[uyl],
+    u.subs({t: tf}) - final_state_constraints[u],
+)
+
+# %%
+# Forcing h > 0.0 sometimes avoids negative 'solutions'. Here it also
+# seem to help with the convergence of the optimization: Different bounds,
+# e.g.: h :math:`\in (10^{-4}, 1.0)` gives unreasonable results.
+
+bounds = {
+    F: (-20., 20.),
+    xl: (starting_location, ending_location),
+    xc: (starting_location, ending_location),
+    h: (0.0, 1.0),
+}
+
+# %%
+# Create the optimization problem.
+prob = Problem(
+    obj,
+    obj_grad,
+    eom,
+    state_symbols,
+    num_nodes,
+    interval_value,
+    known_parameter_map=par_map,
+    instance_constraints=instance_constraints,
+    bounds=bounds,
+)
+
+# %%
+# This initial guess was used to get a solution which forms the actual initial
+# guess:
+#
+i1 = [(ending_location - starting_location)/num_nodes*i for i in range(num_nodes)]
+i2 = [0.0 for _ in range(num_nodes)]
+i3 = i1
+i4 = [-par_map[l] for _ in range(num_nodes)]
+i5 = [0.0 for _ in range(5*num_nodes)]
+i6 = [0.01]
+initial_guess = np.array(i1 + i2 + i3 + i4 + i5 + i6)
+
+# %%
+# The initial guess is the solution of some previous run, saved in the file
+# 'solution.npy', to speed up convergence. To get this solution, the initial
+# guess above was used, which is now overwritten.
+
+initial_guess = np.load('solution.npy')
```

Comment:
Can you rename the file to `crane_moving_a_load_solution.npy` because we will likely add more of these files. I think even another PR has one named the same and they are all in the same directory.

Reply:
> Can you rename the file to `crane_moving_a_load_solution.npy` because we will likely add more of these files. I think even another PR has one named the same and they are all in the same directory.

Makes emminent sense! Will do so right now